c-dependencies/js-compute-runtime/xqd-world/xqd_world.c

```diff
@@ -0,0 +1,2480 @@
+                                        ptr);
+  fastly_result_option_secret_handle_error_t result;
+  switch ((int32_t)(*((uint8_t *)(ptr + 0)))) {
+  case 0: {
+    result.is_err = false;
+    fastly_option_secret_handle_t option;
+    switch ((int32_t)(*((uint8_t *)(ptr + 4)))) {
+    case 0: {
+      option.is_some = false;
+      break;
+    }
+    case 1: {
+      option.is_some = true;
+      option.val = (uint32_t)(*((int32_t *)(ptr + 8)));
+      break;
+    }
+    }
+
+    result.val.ok = option;
+    break;
+  }
+  case 1: {
+    result.is_err = true;
+    result.val.err = (int32_t)(*((uint8_t *)(ptr + 4)));
+    break;
+  }
+  }
+  *ret = result.val.ok;
+  *err = result.val.err;
+  return result.is_err ? 1 : 0;
+}
+
+bool fastly_secret_store_plaintext(fastly_secret_handle_t secret, fastly_option_string_t *ret,
+                                   fastly_error_t *err) {
+  __attribute__((aligned(4))) uint8_t ret_area[16];
+  int32_t ptr = (int32_t)&ret_area;
+  __wasm_import_fastly_secret_store_plaintext((int32_t)(secret), ptr);
+  fastly_result_option_string_error_t result;
+  switch ((int32_t)(*((uint8_t *)(ptr + 0)))) {
+  case 0: {
+    result.is_err = false;
+    fastly_option_string_t option;
+    switch ((int32_t)(*((uint8_t *)(ptr + 4)))) {
+    case 0: {
+      option.is_some = false;
+      break;
+    }
+    case 1: {
+      option.is_some = true;
+      option.val = (xqd_world_string_t){(char *)(*((int32_t *)(ptr + 8))),
+                                        (size_t)(*((int32_t *)(ptr + 12)))};
+      break;
+    }
+    }
+
+    result.val.ok = option;
+    break;
+  }
+  case 1: {
+    result.is_err = true;
+    result.val.err = (int32_t)(*((uint8_t *)(ptr + 4)));
+    break;
+  }
+  }
+  *ret = result.val.ok;
+  *err = result.val.err;
+  return result.is_err ? 1 : 0;
+}
+
+bool fastly_async_io_select(fastly_list_async_handle_t *hs, uint32_t timeout_ms,
+                            fastly_option_u32_t *ret, fastly_error_t *err) {
+  __attribute__((aligned(4))) uint8_t ret_area[12];
+  int32_t ptr = (int32_t)&ret_area;
+  __wasm_import_fastly_async_io_select((int32_t)(*hs).ptr, (int32_t)(*hs).len,
+                                       (int32_t)(timeout_ms), ptr);
+  fastly_result_option_u32_error_t result;
+  switch ((int32_t)(*((uint8_t *)(ptr + 0)))) {
+  case 0: {
+    result.is_err = false;
+    fastly_option_u32_t option;
+    switch ((int32_t)(*((uint8_t *)(ptr + 4)))) {
+    case 0: {
+      option.is_some = false;
+      break;
+    }
+    case 1: {
+      option.is_some = true;
+      option.val = (uint32_t)(*((int32_t *)(ptr + 8)));
+      break;
+    }
+    }
+
+    result.val.ok = option;
+    break;
+  }
+  case 1: {
+    result.is_err = true;
+    result.val.err = (int32_t)(*((uint8_t *)(ptr + 4)));
+    break;
+  }
+  }
+  *ret = result.val.ok;
+  *err = result.val.err;
+  return result.is_err ? 1 : 0;
+}
+
+bool fastly_async_io_is_ready(fastly_async_handle_t handle, bool *ret, fastly_error_t *err) {
```

Comment:
The wpt debug build failed on this function with this error:

> WebAssembly trapped: In func fastly_async_io::is_ready at write is_done: Pointer not aligned to 4: Region { start: 1048374, len: 4 }
> wasm backtrace:
>     0: 0x79b00b - <unknown>!<wasm function 15488>
>     1: 0x25055 - <unknown>!<wasm function 296>
>     2: 0x2dd97 - <unknown>!<wasm function 396>
>     3: 0x809bf1 - <unknown>!<wasm function 18035>
>     4: 0x2d98c - <unknown>!<wasm function 395>

Reply:
Thanks, think I found the issue.